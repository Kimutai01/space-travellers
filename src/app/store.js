import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rocket/rocket';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
});

export default store;
