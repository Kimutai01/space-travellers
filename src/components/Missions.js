import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions, updateMission } from '../redux/missions/missions';

export default function Missions() {
  const allMissions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!allMissions.length) dispatch(fetchMissions());
  }, [dispatch]);

  const handleClick = ({ target }) => {
    const { id } = target;
    dispatch(updateMission(id));
  };

  const activeButton = (
    <button type="button" className="btn btn-primary">
      Active member
    </button>
  );
  const inactiveButton = (
    <button type="button" className="btn btn-secondary">
      Not a member
    </button>
  );

  return (
    <table>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th aria-label="Empty cell" />
        </tr>
      <tbody>
        {allMissions.map((mission) => (
          <tr key={mission.id}>
            <td>
              <h3>{mission.name}</h3>
            </td>
            <td>{mission.description}</td>
            <td>
              {mission.joined && activeButton}
              {!mission.joined && inactiveButton}
            </td>
            <td>
              {mission.joined && (
                <button
                  id={mission.id}
                  type="button"
                  className="btn btn-outline-danger"
                  onClick={handleClick}
                >
                  Leave Mission
                </button>
              )}
              {!mission.joined && (
                <button
                  id={mission.id}
                  type="button"
                  className="btn btn-outline-secondary"
                  onClick={handleClick}
                >
                  Join Mission
                </button>
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}
