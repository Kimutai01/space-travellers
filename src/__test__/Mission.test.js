import Missions from '../components/Missions';
import { Provider } from 'react-redux';
import store from '../app/store';
import TestRenderer from 'react-test-renderer';
test('Mission component testing', () => {
  const TREE = TestRenderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>
  );
  expect(TREE).toMatchSnapshot();
});
