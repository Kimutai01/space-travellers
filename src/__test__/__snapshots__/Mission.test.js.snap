// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component testing 1`] = `
<table>
  <tr>
    <th>
      Mission
    </th>
    <th>
      Description
    </th>
    <th>
      Status
    </th>
    <th
      aria-label="Empty cell"
    />
  </tr>
  <tbody />
</table>
`;
